### PR TITLE
OWASP security verification principles added for Safe Places

### PR DESCRIPTION
**Pull request by AdamLeonSmith**
These are proposed to merge into master.  The principles are derived from OWASP, so probably the most important thing to review is the ones I have listed as not included.  It is accepted that not all of these will be in place for MVP1.

### CODE REVIEW

**Review by altoplano — APPROVED** [review]
LGTM
Was one where it looked like the change was to remove a period at the end of the sentence?
Think I saw an earlier comment that the bullet endings should be consistent (all with, or without periods) doesn't effect security or privacy very directly though, so still approving as is (but if you fix the world wil be a better place.)

**Comment by AdamLeonSmith** [comment]
> LGTM
> Was one where it looked like the change was to remove a period at the end of the sentence?
> Think I saw an earlier comment that the bullet endings should be consistent (all with, or without periods) doesn't effect security or privacy very directly though, so still approving as is (but if you fix the world wil be a better place.)

I will fix this, but let's not let it block the PR :)

File: dynamic_testing/MSTSG_STORAGE.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+[SAF-450](https://pathcheck.atlassian.net/browse/SAF-450): Remove the WRITE_EXTERNAL_STORAGE permission from the AndroidManifest.xml. - Resolution in progress
+
+
+# Test Report
+
+## Testing Local Storage for Sensitive Data (MSTG-STORAGE-1 and MSTG-STORAGE-2)
+
+### Static Review
+
+All of the OWASP principles and checks that were static, e.g code review, were conducted.  The only finding was that the AndroidManifest.xml allows logging to external (insecure) storage, but it is not used.
```
Comment (lmartensson) [inline dynamic_testing/MSTSG_STORAGE.md:49]: Was this issue tracked and fixed?

File: dynamic_testing/MSTSG_STORAGE.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+[SAF-235](https://pathcheck.atlassian.net/browse/SAF-235): Post migration to a secure DB - historical CROSSED_PATHS data remains in the unencrypted DB - Resolution in progress
+
+[SAF-236](https://pathcheck.atlassian.net/browse/SAF-236): Intersection calculation data may be sensitive and is logged on Android - Resolution in progress
+
+[SAF-450](https://pathcheck.atlassian.net/browse/SAF-450): Remove the WRITE_EXTERNAL_STORAGE permission from the AndroidManifest.xml. - Resolution in progress
+
+
+# Test Report
+
+## Testing Local Storage for Sensitive Data (MSTG-STORAGE-1 and MSTG-STORAGE-2)
```
Comment (lmartensson) [inline dynamic_testing/MSTSG_STORAGE.md]: How about including a link to each test description in each section?

File: dynamic_testing/MSTSG_STORAGE.md (hunk 1 above)
Comment (lmartensson) [inline dynamic_testing/MSTSG_STORAGE.md]: Was this issue tracked and fixed? Or what's the plan for the issues you find here? (Could be part of this PR or not, just want to make sure it's clear what the followup will be.)

File: dynamic_testing/MSTSG_STORAGE.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+All of the OWASP principles and checks that were static, e.g code review, were conducted.  The only finding was that the AndroidManifest.xml allows logging to external (insecure) storage, but it is not used.
+
+### Dynamic Review
+
+Secure Database - I wasn't able to gain access to the secure database, but a [process]() has since been identified so this can be conducted in future checks
+
+Legacy Databases - The RKStorage, logback.db and cordova_bg_geolocation.db SQL lite databases were examined.
+
+Issue: CROSSED_PATHS still seems to contained data not migrated to the secure database
```
Comment (lmartensson) [inline dynamic_testing/MSTSG_STORAGE.md]: s/contained/contain/

**Review by lmartensson — APPROVED** [review]
Looks good, only a couple of trivial comments to respond/maybe take simple action on.

File: dynamic_testing/MSTSG_STORAGE.md (hunk 2 above)
Comment (AdamLeonSmith) [reply dynamic_testing/MSTSG_STORAGE.md]: Good idea, note this file is from a different PR

File: dynamic_testing/MSTSG_STORAGE.md (hunk 3 above)
Comment (AdamLeonSmith) [reply dynamic_testing/MSTSG_STORAGE.md]: Agree - but note this file is in a different approved PR

**Pull request merged**